### PR TITLE
API refactor: create

### PR DESCRIPTION
**Pull request by cantsin**
Builds on #89 

Somewhat more complicated due to supporting JSON and CSV through the API (this did not work before). No UI code is touched (it's rather difficult to make the UI call the API due to the presence of the `upload_form_class`). Future API endpoints like `update` and `partial_update` (plus the custom endpoints such as `insert` and `stage`) will be called from the UI, though. Adds a bunch of new tests for the API create endpoint.

### CODE REVIEW

File: data_ingest/tests/test_settings.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
 DATABASES = {
         'default': {
-            'ENGINE': 'django.db.backends.sqlite3',
+            'ENGINE': 'django.db.backends.postgresql',
```
Comment (cantsin) [inline data_ingest/tests/test_settings.py:38]: Changed test database backend to postgres as we are storing JSON data, something not supported with (plain) `sqlite`

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,13 +7,14 @@ jobs:
...
-          DATABASE_URL: postgres://postgres@localhost/data_ingest
+          DATABASE_URL: postgresql://postgres@localhost/test_data_ingest?sslmode=disable
           CI_TESTING: "TRUE"
 
       - image: circleci/postgres:10
         environment:
           POSTGRES_USER: postgres
-          POSTGRES_DB: data_ingest
+          POSTGRES_DB: test_data_ingest
+          POSTGRES_PASSWORD: test_data_password
```
Comment (cantsin) [inline .circleci/config.yml:17]: @amymok alright, so apparently the postgres docker image has a breaking change (ugh) and this breaks our circleCI configuration. turns out, i had to be more explicit with user/password settings. see: https://discuss.circleci.com/t/postgresql-image-password-not-specified-issue/34555

File: data_ingest/utils.py [hunk 3]
```diff
@@ -53,8 +54,21 @@ def to_tabular(incoming):
     are data rows containing data values in the order of headers defined
     in the first row.
     """
+    if incoming.get('source') is None:
```
Comment (amymok) [inline data_ingest/utils.py:57]: If no `source`, how come we don't need to process `incoming`?

Comment (amymok) [reply data_ingest/utils.py:57]: @cantsin can you clarify on this?

Comment (cantsin) [reply data_ingest/utils.py:57]: Great question, this is to provide consistency when we have the case that `source` is nonexistent. We do the exact same thing in `reorder_csv` below. 

To recap, in the `GoodtablesValidator`, we have:

```python
        if content_type == "application/json":
            data = utils.to_tabular(source)
        elif content_type == "text/csv":
            data = utils.reorder_csv(source)
        else:
            raise UnsupportedContentTypeException(content_type, type(self).__name__)
```

IOW, both `to_tabular` and `reorder_csv` can get an empty `source`. Then the validator can proceed as normal.

Comment (cantsin) [reply data_ingest/utils.py:57]: I should mention we also have a test for this! `test_api_validate_json_empty_with_token`

Comment (amymok) [reply data_ingest/utils.py:57]: ah ok, that makes sense.  Thank you.

**Review by amymok — APPROVED** [review]
LGTM!

**Pull request merged**